### PR TITLE
Add tests to report when package references are out of date

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Add tests to report when package references are out of date
  - Also, update the packaging script to generate updated `files.wxs`

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/releaseTools.psm1 [hunk 1]
```diff
@@ -246,32 +251,148 @@ function Get-NewOfficalPackage
...
+{
+    param(
+        [Parameter(Mandatory)]
+        $Version
+    )
+
+    $parts = $Version -split '\.'
+
+    $regEx = "^$($parts[0])\.$($parts[1])\..*"
+    Write-Verbose "version: $Version; regEx: $regEx"
```
Comment (iSazonov) [inline tools/releaseTools.psm1]: Seems the message very short to understand in log where it come from.

File: test/common/package/package.tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+$moduleRootFilePath = Split-Path -Path $PSScriptRoot -Parent
```
Comment (iSazonov) [inline test/common/package/package.tests.ps1]: Please add new line.

File: tools/releaseTools.psm1 [hunk 3]
```diff
@@ -246,32 +251,148 @@ function Get-NewOfficalPackage
...
         # check to see if there is a newer package for each refernce
-        foreach($package in $packages)
+        foreach ($package in $packages)
         {
             # Get the name of the package
             $name = $package.Include
 
             # don't pull 'Microsoft.Management.Infrastructure' from nuget
-            if($name -and $name -ne 'Microsoft.Management.Infrastructure')
+            if ($name -and $name -ne 'Microsoft.Management.Infrastructure')
```
Comment (adityapatwardhan) [inline tools/releaseTools.psm1]: MMI package is on nuget.org.. we should check for that too.

https://www.nuget.org/packages/Microsoft.Management.Infrastructure/

File: tools/releaseTools.psm1 [hunk 4]
```diff
@@ -246,32 +251,148 @@ function Get-NewOfficalPackage
...
+function Test-IncludePackageVersion
+{
+    param(
+        [string]
+        $NewVersion,
+        [string]
+        $Version
+    )
+
+    $simpleCompare = $Version -notlike '*-*'
```
Comment (adityapatwardhan) [inline tools/releaseTools.psm1]: Should we use semantic version instead, would make the checks much simpler. 
```PowerShell
$NewVersion = [System.Management.Automation.SemanticVersion]"6.1.0-preview.4"
$Version = [System.Management.Automation.SemanticVersion]"6.0.4"

if($NewVersion.PreReleaseLabel) { return false } else { return $NewVersion -gt $Version } 
…..

```

File: tools/releaseTools.psm1 [hunk 5]
```diff
@@ -246,32 +251,148 @@ function Get-NewOfficalPackage
...
+    else
+    {
+        # Not sure, include it
+        return $true
+    }
+}
+
+##############################
+#.SYNOPSIS
+# Get a RegEx based on a version tht will match the major and minor
```
Comment (adityapatwardhan) [inline tools/releaseTools.psm1]: Typo tht -> that

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/releaseTools.psm1 (hunk 4 above)
Comment (TravisEz13) [reply tools/releaseTools.psm1]: I had an offline discussion with @adityapatwardhan .  I'll update the synopsis on why the current logic is correct.

**Comment by TravisEz13** [comment]
@adityapatwardhan  @iSazonov  I believe I have addressed your feedback

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@TravisEz13  Please have a look at wix packaging error.

**Review by iSazonov — APPROVED** [review]
Should we update our Contribution guide?

**Comment by TravisEz13** [comment]
This PR is difficult enough to keep up to date.  Could you make a PR to update the contribution guide?

**Comment by iSazonov** [comment]
Done #7695.

**Comment by felixfbecker** [comment]
Does this make the build fail if a new version is released, i.e. irreproducable?

**Comment by TravisEz13** [comment]
First, let us start with why this test exists.  We expect that during development that the latest packages are being used.  Unless there is a good reason, not to, at which time we will adjust the tests, if needed.
During development, we expect the latest revision to a package to be used.  Comments in the code indicate to disable one of the tests until we have a better mechanism for this.

That is two questions combined into one:

1. Does this make the build fail if a new version is released?
    * Yes

1. Is the failure irreproducible?
    * Under certain circumstances the failure would not be reproducible, but that is true for many tests.
      It would not be reproducible if the package owner removed the new package, but the tests logs what the new version was.  
      So, this failure would be diagnosable.

The maintainers are happy to fix this issue, if the issue arises in your PR.

**Pull request merged**